lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -47,6 +53,26 @@ function initializeImportMeta(meta, context, loader) {
   return meta;
 }
 
+/**
+ * Inspect the module URL and convert it to `__dirname` and `__filename`
+ * references if it does not point to a network module.
+ * @param {string} url
+ * @returns {{__dirname?: string, __filename?: string}}
+ */
+function resolveModuleMeta(url) {
+  if (url.slice(0, 7) !== 'file://') {
+    // These only make sense for locally loaded modules,
+    // i.e. network modules are not supported.
+    return { __dirname: undefined, __filename: undefined };
+  }
+
+  const filePath = fileURLToPath(url);
+  return {
+    __dirname: dirname(filePath),
+    __filename: filePath,
```

Comment:
```suggestion
    __dirname: dirname(filePath),
    __filename: filePath,
```

These could probably be lazy since the majority of modules don't use it. It would reduce memory usage and avoid waste.

Reply:
Can you clarify? What change would you make to satisfy your suggestion?

Reply:
made suggestion below